### PR TITLE
Migration: excluded files need to be removed

### LINKED ISSUE #5461: Migration: excluded files need to be removed
If a project.json uses an exclude statement, then the migrated csproj file needs to use Remove. For example:

&lt;Compile Remove="FileToRemove"/&gt;

We need to do the same thing for resources too. Otherwise the migrated project fails to build.

### PR DESCRIPTION
**Pull request by jgoshi**
Fixes #5461 

@piotrpMSFT @livarcocc @jonsequitur @krwq

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/transforms/ConditionalTransform.cs [hunk 1]
```diff
@@ -25,5 +29,74 @@ public U Transform(T source)
...
+                return string.Empty;
+            }
+
+            List<string> mutatedPatterns = new List<string>(patterns.Count());
+
+            foreach (var pattern in patterns)
+            {
+                // Do not use forward slashes
+                // https://github.com/Microsoft/msbuild/issues/724
+                var mutatedPattern = pattern.Replace('/', '\\');
```
Comment (am11) [inline src/Microsoft.DotNet.ProjectJsonMigration/transforms/ConditionalTransform.cs]: Seems like Microsoft/msbuild#724 was fixed by Microsoft/msbuild#1035.

**Comment by piotrpMSFT** [comment]
@jgoshi are you working for a fix here?

**Comment by jgoshi** [comment]
Looking at this now...

**Comment by jgoshi** [comment]
test Windows_NT x64 Release Build

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Comment by jgoshi** [comment]
test Ubuntu16.04 x64 Debug Build

**Comment by jgoshi** [comment]
test Windows_NT x64 Release Build

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Comment by jgoshi** [comment]
test Windows_NT x64 Release Build

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@jgoshi @livarcocc there is a real test failure here. Can you take a look? Kicking out of today's insertion for the moment.

**Comment by piotrpMSFT** [comment]
I'd like to get this in over the weekend, however, if feasible.

**Comment by livarcocc** [comment]
@piotrpMSFT I think that's a genuine failure. Maybe a path length issue.

**Comment by jgoshi** [comment]
This seems like a path length issue. It's complaining about not finding the hostpolicy.dll. Last time I saw that I reduced the size of the test name and the tests passed. So...unless that was by luck, this seems to be a path length issue. Other than that I have no idea how to even investigate this. Will try shortening names one more time.

**Comment by livarcocc** [comment]
I am trying this out now.

**Comment by jgoshi** [comment]
I have a legit test error

**Comment by jgoshi** [comment]
Fixing it now

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build

**Comment by livarcocc** [comment]
@jgoshi Thanks for getting this green.

**Pull request merged**